qadence/backends/api.py

```diff
@@ -44,9 +49,22 @@ def backend_factory(
 
     # Set backend configurations which depend on the differentiation mode
     set_backend_config(backend_inst, diff_mode)
-
+    # Wrap the quantum Backend in a DifferentiableBackend if a diff_mode is passed.
     if diff_mode is not None:
-        backend_inst = DifferentiableBackend(backend_inst, DiffMode(diff_mode))
+        try:
```

Comment:
We'll always need an engine, independent of diffmode no?

Reply:
hm no because you can run things in "no_grad" mode

Reply:
we tell the backend_factory essentially "i want to compute grads" by passing a diff_mode, if none is passed, then torch stuff is run in no_grad mode and for JAX its irrelevant since you only do grads by manually invoking "grad" "value_and_grad" etc. however, if you choose gpsr as diffmode, you need the PSR Expectation which sits in the DiffBackend. lastly,  see "engine" more as an alias for "give me the differentiablebackend for backend "X" which uses autodiff engine "Y")

Reply:
les move this discussion somewhere else if there are some remaining questions